### PR TITLE
Docs: Fix broken link in contribute/backend/README.md

### LINKED ISSUE #92102: Docs: Broken link (404 Errors) to `Upgrading Dependencies` in contribute/backend/README.md
### What happened?

While reading through [contribute/backend/README.me](https://github.com/grafana/grafana/blob/main/contribute/backend/README.md), I received a 404 error when I clicked [Upgrading dependencies](https://github.com/grafana/grafana/blob/main/contribute/backend/upgrade-dependencies.md).

### What did you expect to happen?

To be navigated without a 404 error.

### Did this work before?

Guess not, https://github.com/grafana/grafana/commit/0ecff76600f6830735547c23fb67e5f161c700dc introduced it with broken link

### How do we reproduce it?

Navigate to this [contribute/backend/README.me](https://github.com/grafana/grafana/blob/main/contribute/backend/README.md) and click [Upgrading dependencies](https://github.com/grafana/grafana/blob/main/contribute/backend/upgrade-dependencies.md).

### Is the bug inside a dashboard panel?

No

### Environment (with versions)?

n/a

### Grafana platform?

None

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by rwielk**
**What is this feature?**

Documentation fix, there was a broken link.

**Why do we need this feature?**

To be able to browse documentation more easily.

**Who is this feature for?**

Grafana contributors.

**Which issue(s) does this PR fix?**:

Fixes #92102

**Special notes for your reviewer:**

Please check that:
- [X] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: contribute/backend/README.md [hunk 1]
```diff
@@ -25,4 +25,4 @@ If you want to make or review large changes to the backend, be sure to habituall
 
 ## Guidelines for dependency management
 
-If you work with a dependency that requires an upgrade, refer to [Upgrading dependencies](/contribute/backend/upgrading-dependencies.md).
+If you work with a dependency that requires an upgrade, refer to [Upgrading dependencies](/contribute/backend/upgrade-dependencies.md).
```
Comment (hairyhenderson) [inline contribute/backend/README.md:28]: link name should change too to reflect the title:

```suggestion
If you work with a dependency that requires an upgrade, refer to [Upgrade dependencies](/contribute/backend/upgrade-dependencies.md).
```

**Review by hairyhenderson — COMMENTED** [review]
Thanks for this @rwielk! Just one small change then it can merge.

**Review by hairyhenderson — APPROVED** [review]
LGTM, thanks @rwielk!

**Comment by rwielk** [comment]
@hairyhenderson Would you mind to have a look at drone ci ? I miss the rights to restart job, got stuck there, no idea why tbh

**Comment by irenerl24** [comment]
hey, as we're having trouble with CI in this PR, I've created this one (#93222) to fix this broken link. Thanks for contributing!

**Pull request merged**